site/docs/view-spec.md

```diff
@@ -0,0 +1,256 @@
+| Required | representations | A list of "representations" as described below. |
+
+
+Note that each version is stored in a separate AVRO file. This is to ensure that the metadata file stays readable in the case the view definition is huge. As a future extension, an engine-agnostic intermediate representation or a serialized abstract syntax tree of the SQL definition may also be stored in each version, exacerbating the problem.
+
+“summary” is a string-string map with the following required keys. Engines may store additional key-value pairs in this map.
+
+| Required/Optional | Key | Value |
+|-------------------|-----|-------|
+| Required | operation |A string value indicating the view operation that caused this metadata to be created. Allowed values are “CREATE” and “REPLACE” |
+| Optional | engine-version | A string value indicating the version of the engine that performed the operation (create / replace) |
+
+“representations” is a list of structs with fields as shown below:
+
+| Required/Optional | Field Name | Description |
+|-------------------|------------|-------------|
+| Required | type | A string indicating the type of representation. The only valid choice is "sql". The rest of the fields are interpreted by the type of representation. |
+
+Here are the fields for "sql" representation type:
+
+| Required/Optional | Field Name | Description |
+|-------------------|------------|-------------|
+| Required | type | It must be set to "sql" |
+| Required | sql | A string representing SQL definition of the view as input |
+| Required | dialect | A string specifying the dialect of the ‘sql’ field above. Used by engines to perform necessary translations to the SQL dialect supported by the engine. |
+| Optional | session-catalog | A string that specifies the catalog of the user session when the view was created / replaced. Used to resolve the tables in the view definition. |
+| Optional | session-namespace | An array of strings indicating namespace at the time view was created / replaced. Used similar to ‘session-catalog’ above. |
+| Optional | field-aliases | A list of strings of field aliases E.g. a list of alias_name info specified in the following create view statement. `CREATE VIEW v (alias_name COMMENT 'docs', alias_name2, ...) AS SELECT ...` |
+| Optional | field-docs | A list of strings of field comments E.g. a list of ‘comment’ info specified in the following create view statement. `CREATE VIEW v (alias_name COMMENT 'docs', alias_name2, ...) AS SELECT ...` |
+
+“version-log” is an array of structs describing the log of the versions created. The struct has the following fields:
+
+| Required/Optional | Field Name | Description |
+|-------------------|------------|-------------|
+| Required | timestamp-ms | Timestamp expressed as ms since epoch for when the version of the view indicated in version-id field below was created |
```

Comment:
This isn't correct. This timestamp should be when the referenced version was made the current version.